vnext/Microsoft.ReactNative/XamlLoadState.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+// Turn this off if you'd like to disable XAML versioning guard.
+#define XAML_GUARD
+
+/// <summary>
+/// Implements a guard to ensure that only one XAML dll gets loaded in the current process.
+/// This is important because different XAML dlls are not inter-operable.
+/// It relies on APIs and types that are part of the WDK and not usually available via the UWP SDKs,
+/// but the docs allow calling these APIs via GetProcAddress.
+/// </summary>
+struct XamlLoadState {
+  enum class XamlDialect {
+    Unknown = -1,
+    SystemXAML = 1,
+    WinUI = 2,
+  };
+
+  struct XamlVersion {
+    uint16_t m_major{};
+    uint16_t m_minor{};
+    std::wstring m_path{};
+    XamlDialect GetMode() const noexcept;
+    XamlDialect GetKnownMode() const;
+  };
+
+  XamlLoadState();
+  ~XamlLoadState();
+
+#ifdef XAML_GUARD
+  static XamlLoadState g_Instance;
+#endif
+
+  void RegisterDll(PCWSTR DllName, PCWSTR path);
```

Comment:
>) [](start = 46, length = 1)

Could it it be noexcept?

Reply:
no, it relies on exceptions to catch invalid xaml dialect combinations.